spring-integration-core/src/main/java/org/springframework/integration/config/annotation/AbstractMethodAnnotationPostProcessor.java

```diff
@@ -272,6 +273,10 @@ protected void configurePollingEndpoint(AbstractPollingEndpoint pollingEndpoint,
 			String fixedDelayValue = this.environment.resolvePlaceholders(poller.fixedDelay());
 			String fixedRateValue = this.environment.resolvePlaceholders(poller.fixedRate());
 			String maxMessagesPerPollValue = this.environment.resolvePlaceholders(poller.maxMessagesPerPoll());
+			if (pollingEndpoint instanceof SourcePollingChannelAdapter) {
+				// SPCAs default to 1 message per poll
+				maxMessagesPerPollValue = "1";
```

Comment:
Doh - I had that originally - I first of all passed in a boolean to indicate a inbound Vs. outbound but then I realized I could test the adapter type; I lost the StringUtils call when I refactored.

Thanks for catching it!